src/v2/Apps/Order/Components/__tests__/AddressModal.jest.tsx

```diff
@@ -24,40 +25,112 @@ const savedAddress: SavedAddressType = {
   isDefault: false,
 }
 
-const testAddressModalProps: Props = {
-  show: true,
-  address: savedAddress,
-  onSuccess: jest.fn(),
-  onError: jest.fn(),
-  onDeleteAddress: jest.fn(),
-  modalDetails: {
-    addressModalTitle: "Modal title",
-    addressModalAction: "editUserAddress",
-  },
-  me: {
-    id: "1234",
-    addressConnection: {
-      edges: [],
-    },
-    " $refType": "SavedAddresses_me",
-  },
-  closeModal: jest.fn(),
-}
+let testAddressModalProps: Props
 
 function getWrapper(props: Props) {
   return mount(<AddressModal {...props} />)
 }
 
 describe("AddressModal", () => {
   beforeEach(() => {
+    testAddressModalProps = {
+      show: true,
+      address: savedAddress,
+      onSuccess: jest.fn(),
+      onError: jest.fn(),
+      onDeleteAddress: jest.fn(),
+      modalDetails: {
+        addressModalTitle: "Edit address",
+        addressModalAction: "editUserAddress",
+      },
+      me: {
+        id: "1234",
+        addressConnection: {
+          edges: [],
+        },
+        " $refType": "SavedAddresses_me",
+      },
+      closeModal: jest.fn(),
+    }
     commitMutation.mockReset()
   })
-  it("renders Modal with the title and input fields", () => {
+  it("renders EditModal with the title, input fields and buttons", () => {
     const wrapper = getWrapper(testAddressModalProps)
-    expect(wrapper.text()).toContain("Modal title")
+    expect(wrapper.text()).toContain("Edit address")
+    expect(wrapper.find("input").length).toBe(7)
+    expect(wrapper.find("select").length).toBe(1)
+    expect(wrapper.find("Text[data-test='deleteButton']").length).toBe(1)
+    expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
+  })
+
+  it("renders AddModal with the title, input fields and button", () => {
+    const wrapper = getWrapper({
+      ...testAddressModalProps,
+      modalDetails: {
+        addressModalTitle: "Add address",
+        addressModalAction: "createUserAddress",
+      },
+    })
+    expect(wrapper.text()).toContain("Add address")
     expect(wrapper.find("input").length).toBe(7)
     expect(wrapper.find("select").length).toBe(1)
+    expect(wrapper.find("Text[data-test='deleteButton']").length).toBe(0)
```

Comment:
non-blocking:

This particular line is the only unique assertion in this test; everything else is also covered by the previous test. I personally find it helpful to remove setup & assertions that aren't unique to a test, so that when someone is reading this test in the future, they can quickly identify the unique scenario it is testing. 

I'm not suggesting you change it for this PR to be merged; just pointing out that duplication in tests like this can mask the intention of the test.